### PR TITLE
#161779866 implement create user profile functionalities

### PR DESCRIPTION
**Pull request by ekpangmichael**
#### What does this PR do?
Implement create, read and update on user profile
#### Description of Task to be completed?
- I created a user profile controller
- I added endpoints to route
- I protect routes with JWT
- Wrote test to cover profile endpoints
#### How should this be manually tested?
- Clone the repo
- Cd into the directory
- Run `` npm insall``
- Run `` npm run test ``
Test the following routes
``` GET  api/v1/user/profile/ ```
``` GET  api/v1/user/profile/:id ```
``` PUT  api/v1/user/profile/ ```

#### Any background context you want to provide?
You will need to hit the user signin route to get the token
Token value should follow this format ``` Bearer Token```

#### What are the relevant pivotal tracker stories?
#161779866
#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: server/controllers/userProfileController.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
+import db from '../models';
+
+const { Profile } = db;
+/**
+* @description class will implement users profile functionality
+*
+* @class userProfileController
+*/
+class userProfileController {
```
Comment (davidshare) [inline server/controllers/userProfileController.js]: Well done Michael. Please class names should start with a capital letter.

File: server/controllers/userProfileController.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+        bio: getUser.bio,
+        twiiter: getUser.twitterUrl,
+        facebook: getUser.facebookUrl,
+        role: getUser.role,
+        image: getUser.imageUrl,
+        gender: getUser.gender,
+      };
+
+      return res.status(200).json({
+        status: 'Success',
```
Comment (davidshare) [inline server/controllers/userProfileController.js]: Please the status code has already been sent, it should not be sent in the response object. Instead,  set success to true in the response body.

**Review by idmega2000 — APPROVED** [review]
Nice one Michael. I like the fact that you create a validateUUID middleware that validates uuid. This middleware can be used by any other person who wants to work directly with uuid on the project. 
Nice job.

**Review by Danielshow — APPROVED** [review]
Looks good to me

File: server/middlewares/UuidValidator.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import checkUUID from './validations/validationHelper';
```
Comment (efalayi) [inline server/middlewares/UuidValidator.js]: @ekpangmichael please rename this file to `uuidValidator.js`

Comment (ekpangmichael) [reply server/middlewares/UuidValidator.js]: Okay

**Review by efalayi — APPROVED** [review]

**Pull request merged**